Assets/Scripts/Framework/StateMachine/StateMachineManger.cs

```diff
@@ -0,0 +1,68 @@
+using UnityEngine;
+
+public abstract class StateMachineManger : MonoBehaviour
+{
+    [Header("Base StateMachine")]
+    [SerializeField] protected BaseState startingState;
+    [SerializeField] protected BaseState[] states;
+    
+    private static BaseState _currentState;
+
+    private void Update() => _currentState.UpdateState(this);
+    private void FixedUpdate() => _currentState.FixedUpdateState(this);
+
+    /// <summary>
```

Comment:
moet dit er nog in blijven?

Reply:
Ja, het geeft uitleg over de functie er over.